### DISCUSSION TITLE
SL4 AMD 15" PopOS 21.04 w/LUKS - My experience

### DISCUSSION THREAD

**Discussion by professor-rage** [linux-surface/linux-surface#584] [General]
First of all, thanks to the @qzed @tmsp for enabling Linux support on Surface hardware. My SL4 AMD 15" arrived this week! Here are my notes installing/configuring PopOS 21.04.

### PopOS 21.04 w/LUKS and kernel 5.14.8-surface
Everything seems stable minus suspend. 

### PopOS Install/configuration:
1. Until the linux-surface kernel is installed, connect an external keyboard and mouse because the laptop keyboard and trackpad will not work.  
2. Hold the F2 key when turning on the laptop to boot from the PopOS Live USB drive
3. Hit the escape key to add the following boot/kernel params: 
```
amd_iommu=force_isolation acpi_backlight=vendor
```

### Setting kernel params permanently in PopOS:
Enables brightness control and prevent freezing on boot:
```
sudo kernelstub -a "amd_iommu=force_isolation"
sudo kernelstub -a "acpi_backlight=vendor"
```

### Enable surface keyboard in LUKS interface
Once the linux-surface kernel is installed and active, edit /etc/initramfs-tools/modules and add the following to the end of the file:
```
surface_aggregator
surface_aggregator_registry
surface_hid_core
surface_hid
8250_dw
```

###Update initramfs
```
sudo update-initramfs -k all -u
```
### Suspend issues/quirks/notes
1. Power draw in suspend is high enough to deplete fully charged battery overnight.
2. In suspend, the keyboard backlight stays on 
3. To wake from suspend, disconnect or connect the surface power cable. The keyboard and trackpad will not wake.

I don't know enough about Kernel dev, ACPI, S0ix/s2Idle etc to help with the solution, but I can test with step-by-step instructions. 

Thanks

**Comment by qzed** [comment]
Suspend is very much a work in progress and doesn't work correctly yet. We're tracking that in #458, @nakato is doing some excellent work to improve the situation.

**Reply by professor-rage** [reply]
Thanks. Seems like suspend is a work in progress for everyone these days. I tried the Dell 9710 and Asus M16 with Tiger Lake 11th gen H processors. Both had high power drain in suspend with Debian-based 20.04/21.04. Even in Windows 10, I eventually landed on the following: 
https://www.reddit.com/r/System76/comments/k7xrtz/ill_have_whatever_intel_was_smoking_when_they/

With my basic understanding (or mis-understanding), it seems like Intel/AMD/standards groups are trying to get more intelligent with power management by moving away from traditional suspend, but as a result, the user experience to preserve battery when inactive has regressed to 1.shutting down the laptop 2. using inefficient hibernate or 3. always keeping the device plugged in. Maybe I got all of this wrong and suspend has always been a challenge when new hardware comes out. If you have a good link that breaks any of this down, please share.

**Reply by qzed** [reply]
Yeah, the move is to S2idle/"Connected Standby" which means you can e.g. download updates (or do other fancy things) while the device is "sleeping" (for all that's worth). IIRC the Nintendo Switch is also using that for updates, at least when docked. What I really don't get is why that has to be an either S3 _or_ S2idle/CS thing... just let users decide and support both (might IMHO even be a better user experience if you switch from S2idle to S3 after x hours to save battery). Claims are that it has "similar power-savings compared to S3" but I seriously doubt that.

Unfortunately I don't have a good link besides https://01.org/blogs/qwang59/2018/how-achieve-s0ix-states-linux which deals more with the S0ix "active" part. Although S2idle is basically more or less forced S0ix entry (if it works and no device/component is blocking it; if anything is blocking it you're not going into S0ix but still seem suspended...), on Intel devices at least.

**Comment by professor-rage** [comment]
Just confirming that 5.14.9 fixed the excessive power draw in "suspend" and idle power draw is lower as well. 

The only remaining issue is coming out of suspend. The trackpad and keyboard will not wake. Instead, plugging in a USB device or connecting the power cable will do the job.

**Reply by qzed** [reply]
Yeah, keyboard and touchpad won't work as a wakeup source. Opening/closing the lid should work though. Hopefully with the next release the power button will work as well.

**Comment by PeterDragon50** [comment]
I know this post was long ago, but I was wondering if someone could help a relative noob out with this? I cannot get my SL4 R5 13.5 to reliably bring up the are to edit the Params. I am pressing F2 while turning on the device, I am pressing Esc when it goes to the "Try or Install screen" and nothing. I somehow got it to do this once, but the Params did not take, likely because I entered them incorrectly. 

Any help would be more than appreciated.
